runelite-client/src/main/java/net/runelite/client/plugins/combatlevel/CombatLevelOverlay.java

```diff
@@ -0,0 +1,151 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.combatlevel;
+
+import net.runelite.api.Client;
+import net.runelite.api.Skill;
+import net.runelite.api.TabIndex;
+import net.runelite.api.VarClientInt;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.tooltip.Tooltip;
+import net.runelite.client.ui.overlay.tooltip.TooltipManager;
+import javax.inject.Inject;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+
+public class CombatLevelOverlay extends Overlay
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private CombatLevelConfig config;
+
+	@Inject
+	private TooltipManager tooltipManager;
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showLevelsUntil() || client.getLocalPlayer().getCombatLevel() == 126
+			|| client.getVar(VarClientInt.CURRENT_TAB) != TabIndex.ATTACK_STYLES.getValue())
+		{
+			return null;
+		}
+
+		Widget combatLevelWidget = client.getWidget(WidgetInfo.COMBAT_LEVEL);
+
+		if (combatLevelWidget == null)
+		{
+			return null;
+		}
+
+		Rectangle combatCanvas = combatLevelWidget.getBounds();
+
+		if (combatCanvas == null)
+		{
+			return null;
+		}
+		
+		if (combatCanvas.contains(client.getMouseCanvasPosition().getX(), client.getMouseCanvasPosition().getY()))
+		{
+			tooltipManager.add(new Tooltip(getLevelsUntil()));
+		}
+
+		return null;
+	}
+
+	private String getLevelsUntil()
+	{
+		// put together information
+		int attackLevel = client.getRealSkillLevel(Skill.ATTACK);
+		int strengthLevel = client.getRealSkillLevel(Skill.STRENGTH);
+		int defenceLevel = client.getRealSkillLevel(Skill.DEFENCE);
+		int hitpointsLevel = client.getRealSkillLevel(Skill.HITPOINTS);
+		int magicLevel = client.getRealSkillLevel(Skill.MAGIC);
+		int rangedLevel = client.getRealSkillLevel(Skill.RANGED);
+		int prayerLevel = client.getRealSkillLevel(Skill.PRAYER);
+
+		double base = 0.25 * (defenceLevel + hitpointsLevel + Math.floor(prayerLevel / 2));
+		double melee = 0.325 * (attackLevel + strengthLevel);
+		double range = 0.325 * Math.floor(rangedLevel * 1.5);
+		double mage = 0.325 * Math.floor(magicLevel * 1.5);
+		double max = Math.max(melee, Math.max(range, mage));
+
+		// begin calculating levels
+		int next = client.getLocalPlayer().getCombatLevel() + 1;
+		int meleeNeed = calcLevels(base + melee, next, 0.325);
+		int hpdefNeed = calcLevels(base + max, next, 0.25025);
+		int prayNeed = calcLevels(base + max, next, 0.125);
+		int rangeNeed = calcLevelsRM(rangedLevel, next, base);
+		int magicNeed = calcLevelsRM(magicLevel, next, base);
+
+		if (prayerLevel % 2 == 0)
+		{
+			prayNeed++;
```

Comment:
Just wanted to point out, I've noticed that https://oldschool.tools/calculators/combat-level also performs this check, and it incorrectly overestimates the number of prayer levels needed. I think the correct adjustment is `if ((prayerLevel + prayNeed) % 2 == 1) prayNeed--;`. Raising prayer to an odd level will never increase combat level.